lab_2_tokenize_by_bpe/main.py

```diff
@@ -47,6 +85,14 @@ def merge_tokens(
     :param pair: a pair of tokens to be merged
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(word_frequencies, dict) or not isinstance(pair, tuple):
+        return None
+    merged_dict = {}
+    for word, freq in word_frequencies.items():
+        new_word = [pair[0] + pair[1] if (word[i], word[i + 1]) == pair else word[i] for i in range(len(word) - 1)]
```

Comment:
will `pair[1]` be duplicated?

Reply:
As I see, no, because the part of code replaces the occurrence of the pair with a merged token, which is like the union of pair[0] and pair[1], new token is added  without including pair[1] again